src/pybind/mgr/cephadm/serve.py

```diff
@@ -888,6 +888,13 @@ def update_progress() -> None:
                         hosts_altered.add(d.hostname)
                         break
 
+                # do not attempt to deploy node-proxy agent when oob details are not provided.
+                if slot.daemon_type == 'node-proxy' and not slot.hostname in self.mgr.node_proxy_cache.oob.keys():
+                    self.log.debug(
+                        f'Not deploying {slot.daemon_type} agent on {slot.hostname} as oob details are not present.'
```

Comment:
NIT: isn't `{slot.daemon_type}` always just going to be `node-proxy` here? Is there any reason to have it be a variable in the message rather than just be `node-proxy` directly?